### DISCUSSION TITLE
关于缓存的问题，缓存里缓存了fakeip，但是我并不想他缓存fakeip，如何是好？

### DISCUSSION THREAD

**Discussion by ccczj2323** [IrineSistiana/mosdns#817] [Q&A]
# 缓存
  - tag: cache_nofakeip
    type: cache
    args:
      size: 131072
      lazy_cache_ttl: 3600
      cache_everything: true
      lazy_cache_reply_ttl: 5

  # cache 的 fakeip抛弃
  - tag: "cache_and_check_fakeip"
    type: "sequence"
    args:
      - exec: cache_nofakeip    # 缓存
      - matches:                # 匹配条件。多个匹配条件为 AND 关系。
        - "resp_ip 198.18.0.1/16"
        - "has_resp"
        exec: drop_resp         # 如返回的 IP 为fakeip，则抛弃。

我的解决办法是上面这样，但是实际运行下来会报错，我应该怎么修改呢？

**Comment by yyysuo** [comment]
简单点可以套娃，几个mosdns套起来，上游的mosdns开缓存就行了。

**Comment by Jio0oiJ** [comment]
把fakeip的解析放在缓存命中之前。

**Comment by sam452167735789** [comment]
需要这样，- matches "!xxx"; exec: cache

**Comment by iceyheart** [comment]
- matches: 
          - qname $proxy_list
          exec: $forward_proxy
        - matches: has_resp
          exec: accept
        - exec: $cache
        - matches: has_resp
          exec: accept
        - exec: $forward_direct
        - matches: has_resp
          exec: accept
如上的顺序，就不会去cache查找。
